adafruit_esp32spi/adafruit_esp32spi_socket.py

```diff
@@ -148,6 +151,39 @@ def settimeout(self, value):
         """Set the read timeout for sockets, if value is 0 it will block"""
         self._timeout = value
 
+    def available(self):
+        """Returns how many bytes of data are available to be read (up to the MAX_PACKET length)"""
+        if self.socknum != NO_SOCKET_AVAIL:
+            return min(_the_interface.socket_available(self._socknum), MAX_PACKET)
+        return 0
+
+    def connected(self):
+        """Whether or not we are connected to the socket"""
+        if self.socknum == NO_SOCKET_AVAIL:
+            return False
+        elif self.available():
+            return True
+        else:
+            status = _the_interface.socket_status(self.socknum)
+            # TODO: why is esp.<ConstantName> not defined? using magic numbers in mean time
+            result = status not in (1,
```

Comment:
for context, stole this logic from the arduino `WiFiNINA` lib: https://github.com/arduino-libraries/WiFiNINA/blob/master/src/WiFiClient.cpp#L221-L242